### PR TITLE
DPC-4091: Log AO Verification Failures

### PR DESCRIPTION
**Pull request by jdettmannnava**
## 🎫 Ticket

https://jira.cms.gov/browse/DPC-4091

## 🛠 Changes

- [audited](https://github.com/collectiveidea/audited) gem added
- logging added to verification jobs
- logging added to AO registration failure due to failed CPI API Gateway check
- auditing added to changes to verification status for User, AoOrgLink, and ProviderOrganization
- log when Org access to DPC cut off due to verification failure
- moved logger configuration to application.rb

## ℹ️ Context

- Logging: We want to have insights into where and how often AO Verification fails
- Auditing: We want to have an audit trail when removing access to DPC
- Move of logger configuration because in previous system Rails.logger != logger

## 🧪 Validation

Manual and automated testing

### CODE REVIEW

File: dpc-portal/spec/jobs/verify_provider_organization_job_spec.rb [hunk 1]
```diff
@@ -85,39 +85,95 @@
       end
     end
     context :failures do
-      it 'should update org and link if org has med sanctions' do
-        provider_organization = create(:provider_organization, last_checked_at: 8.days.ago, npi: '3598564557',
-                                                               verification_status: :approved)
-        links = []
-        3.times do
-          links << create(:ao_org_link, provider_organization:)
+      def expect_log_for(link, reason)
```
Comment (chris-ronning-ny) [inline dpc-portal/spec/jobs/verify_provider_organization_job_spec.rb:88]: Oh that's neat, cool stuff

File: dpc-portal/app/models/provider_organization.rb [hunk 2]
```diff
@@ -72,6 +74,12 @@ def disable_rejected
 
       logger.error(['CredentialAuditLog failure',
                     { action: :remove, credential_type: :client_token, dpc_api_credential_id: token['id'] }])
-    end
+    end.present? && log_disabled # Only log if has tokens
+  end
+
+  def log_disabled
+    logger.info(['Org API disabled',
+                 { actionType: LoggingConstants::ActionType::ApiBlocked,
```
Comment (chris-ronning-ny) [inline dpc-portal/app/models/provider_organization.rb:82]: No actionContext?

**Review by chris-ronning-ny — COMMENTED** [review]
Looks reasonable to me for the most part. See a few comments.

File: dpc-portal/config/initializers/logging.rb [hunk 3]
```diff
@@ -2,19 +2,11 @@
...
 
-Rails.application.configure do
-  unless ENV['DISABLE_JSON_LOGGER'] == 'true'
-    Rails.logger = DpcJsonLogger.new($stdout)
-    config.logger = Rails.logger
-    config.logger.formatter = DpcJsonLogger.formatter
-    config.log_formatter = DpcJsonLogger.formatter
-  end
-end
-
```
[the remark this answers is not in the document]
Comment (jdettmannnava) [reply dpc-portal/config/initializers/logging.rb:13]: I don't mind it here. @kyeah ?

File: dpc-portal/app/models/provider_organization.rb (hunk 2 above)
Comment (jdettmannnava) [reply dpc-portal/app/models/provider_organization.rb]: Added if I can get it!

File: dpc-portal/config/initializers/logging.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (chris-ronning-ny) [reply dpc-portal/config/initializers/logging.rb:13]: Yeah I think it's okay too. I'm generally interested in learning Rails best practices, and wanted to make sure we revisited this decision I made a few PRs ago.

**Review by chris-ronning-ny — APPROVED** [review]
Looks good, thanks Jeff!

**Pull request merged**